### PR TITLE
PLUGIN-1637 PLUGIN-1638 PLUGIN-1639 - SFMC changes

### PR DESCRIPTION
**Pull request by sgarg-CS**
[PLUGIN-1637](https://cdap.atlassian.net/browse/PLUGIN-1637)
[PLUGIN-1638](https://cdap.atlassian.net/browse/PLUGIN-1638)
[PLUGIN-1639](https://cdap.atlassian.net/browse/PLUGIN-1639)

Add validation for Data Extension External Key, column mapping, handle spaces in Data Extension External Key, Remove Service.asmx from SOAP API Endpoint URL and refactor tooltips

[PLUGIN-1637]: https://cdap.atlassian.net/browse/PLUGIN-1637?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[PLUGIN-1638]: https://cdap.atlassian.net/browse/PLUGIN-1638?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[PLUGIN-1639]: https://cdap.atlassian.net/browse/PLUGIN-1639?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/java/io/cdap/plugin/sfmc/source/MarketingCloudSourceConfig.java [hunk 1]
```diff
@@ -355,4 +375,22 @@ private void  validateFilter(FailureCollector collector) {
...
+      MarketingCloudClient marketingCloudClient = MarketingCloudClient.getOrCreate(getConnection().getClientId(),
+                                                                                   getConnection().getClientSecret(),
+                                                                                   getConnection().getAuthEndpoint(),
+                                                                                   getConnection().getSoapEndpoint());
+
+      ETResponse<ETDataExtension> response = marketingCloudClient.retrieveDataExtensionKeys();
+      actualDataExtensionKeys = response.getObjects().stream().map(e -> e.getKey())
+        .collect(Collectors.toList());
+    } catch (ETSdkException e) {
+      throw new RuntimeException();
```
Comment (MrRahulSharma) [inline src/main/java/io/cdap/plugin/sfmc/source/MarketingCloudSourceConfig.java]: We should not throw generic RuntimeException, instead we should use the collector to collect the details about this failure.

File: src/main/java/com/custom/fuelsdk/PaginationETSoapObject.java [hunk 2]
```diff
@@ -78,24 +78,28 @@ public static ETResponse<ETDataExtensionRow> select(ETClient client,
                                                       String dataExtension,
```
Comment (MrRahulSharma) [inline src/main/java/com/custom/fuelsdk/PaginationETSoapObject.java:79]: Please add function comments.

File: src/main/java/io/cdap/plugin/sfmc/source/MarketingCloudSourceConfig.java (hunk 1 above)
Comment (sgarg-CS) [reply src/main/java/io/cdap/plugin/sfmc/source/MarketingCloudSourceConfig.java]: Added failure details to the collector. e640c5ddd290ea289079ab0e05ac1a45de8d97ff

File: src/main/java/com/custom/fuelsdk/PaginationETSoapObject.java (hunk 2 above)
Comment (sgarg-CS) [reply src/main/java/com/custom/fuelsdk/PaginationETSoapObject.java:79]: Added comments to the function. e640c5ddd290ea289079ab0e05ac1a45de8d97ff

Comment (MrRahulSharma) [reply src/main/java/com/custom/fuelsdk/PaginationETSoapObject.java:79]: Add some details explaining what this method is supposed to do, what the dataExtension variable expected to be.

File: src/main/java/io/cdap/plugin/sfmc/source/MarketingCloudSourceConfig.java [hunk 3]
```diff
@@ -355,4 +375,30 @@ private void  validateFilter(FailureCollector collector) {
...
+                                                                                   getConnection().getAuthEndpoint(),
+                                                                                   getConnection().getSoapEndpoint());
+
+      ETResponse<ETDataExtension> response = marketingCloudClient.retrieveDataExtensionKeys();
+      actualDataExtensionKeys = response.getObjects().stream().map(e -> e.getKey())
+        .collect(Collectors.toList());
+    } catch (ETSdkException e) {
+      collector.addFailure("Unable to connect to Salesforce Instance.",
+                           "Ensure properties like Client ID, Client Secret, API Endpoint " +
+                             ", Soap Endpoint, Auth Endpoint are correct.")
```
Comment (MrRahulSharma) [inline src/main/java/io/cdap/plugin/sfmc/source/MarketingCloudSourceConfig.java]:
```suggestion
                           "Ensure properties like Client ID, Client Secret, API Endpoint, " +
                             "Soap Endpoint, Auth Endpoint are correct.")
```

**Review by MrRahulSharma — APPROVED** [review]

File: src/main/java/com/custom/fuelsdk/PaginationETSoapObject.java (hunk 2 above)
Comment (sgarg-CS) [reply src/main/java/com/custom/fuelsdk/PaginationETSoapObject.java:79]: Added. 005d9537d6d0e04d265e14d3c4670fa3bbbe140a

**Review by Aryan-Verma — APPROVED** [review]

**Pull request merged**